---
title: "Report"
author: "Khalida Dushimova, Richard Efraim Langi, Madleen Piegsa, Greta Karathanos"
date: '2022-07-12'
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


1. Introduction

RNA-binding proteins are proteins that bind to the double or single stranded RNA in cells and by this interact with transcripts of RNA-driven processes. This large group of proteins plays the key roles in RNA processing and modification (Alternative splicing, RNA editing and polyadenylation), export, mRNA localization and translation. 
The malfunction of RBP´s underlies the origin of many diseases from muscular atrophies and neurological disorders to cancer. (Referenz) Therefore, it is necessary to increase the number of recognized RBPs and the understanding of their molecular mechanisms. Since identifying molecular mechanisms is one of the biggest challenge in (nc)RNA (-> non coding), research M.Caudron-Herger et al. created R-Deep with the aim to detect RBPs automatically. In general, R-Deep is a proteome-wide, unbiased, and enrichment-free screen based on density gradient ultracentrifugation. Performing R-Deep, lysates with a previous RNase treatment (RNase group) and without a previous RNase treatment (Ctrl group) were separated by their density. Each fraction was analyzed by mass spectometry or western blot according to their protein amount. True-positive detected RNA-dependent proteins can bind directly to the RNA (RBP) or bind to RBPs. 

2. Materials and Methods

2.1 RDeep:
R-DeeP is a proteome-wide, unbiased and enrichment-free screen. The principle of R-DeeP is based on cellular lysate fractionation by density gradient ultracentrifugation. (Referenz) The outcome is analyzed by proteome-wide mass spectometry or individual western blotting. In general, R-DeeP is used to determine RNA-dependent protein, that interact directly or indirectly with RNA. Lysates with (RNase group) and without (Ctrl group) RNase treatment are compared. By that differences in molecular weight, hence and the size of the complexes are determined.RBPs are expected to split after RNase treatment and migrate to different fractions in a sucrose density gradient.

```{r}
![](images/.png)
```



2.2 Dataset exploration:
Our given dataset consists of mass-spectometry data from non-synchronized A549 cells which are human lung carcinoma cells of a caucasian male. In general, the data show the protein amount of each of our 3680 human proteins per fraction. The RDeep screen has been repeated three times so it comprises three replicates for each sample. All in all, we got 3680 rows with one human proteins per row and 150 columns for our Ctrl and RNase group for 25 fractions and three replicates each. 



2.3 Fractionwise Normalization:
The total amount of protein of every replicate should be similar. In our given dataset this was not the case. Normalization is the process to account for the bias and make samples more comparable. Our aim was to change the values of columns to a common scale without distorting differences in the range of values and by this reducing the variation between our three technical replicates.
Therefore, for our first normalization we compute the sums fractionwise and find the two closest sums. We then define normalization factors for the replicates. The quotients of the mean sums and sums of replicates are the normalization factors. 

2.4 Anti-outlier function:
For our second normalization, we defined the mean out of the two most similar replicates and by this wrote an own function to exclude outliers. The outliers were found, removed, and replaced by the mean of the other two most similar replicates. 


2.5 Peaks identification:

Peaks We expect RNA dependent proteins to migrate to migrate to different positions in the RNase treated sample compared to the Ctrl sample. As a next step we identify the maxima in both samples to characterize a protein as RNA dependent or RNA independent. We detect the local and the global maxima to draw all the biological information we could possibly get from our given values.
Our theoretical maximum is a point x whose neighboring values have to be smaller. 

The method is based on checking the neighbours of each fraction. For the first fraction, only the right neighbour will be compared because there is no left neighbour. For the second till twenty fourth fraction, both left and right neighbours will be compared. For the last fraction, only the left neighbour will be compared since right neighbour doesn't exist. If the the checked fraction have higher values than the neighbour(s), then the fraction is a maxima. If the the checked fraction have lower values than the neighbour(s), then the fraction is a not a maxima and will be zero-ed in our code.

```{r}
#(Bild selbstgemalt Maxima detection)

```

2.6 Criteria for RNA dependency:

2.6.1 T-Test of absolute maxima:

T-Test of global maxima
T-Test shows us if the global maxima of each of the proteins (all replicates) between Ctrl and RNase group has a significant difference. We declare a significant difference between both groups as RNA dependent. It will be our first criteria for RNA dependency.


2.6.2 K-Means Clustering (Y-shift and X-shift detection):

K-Means Clustering is an unsupervised non-linear algorithm that clusters data based on similarity. Its aim is to partition the observations into a previously definded number of clusters. We define this number by performing the elbow method. 
Through K-Means Clustering and comparing our t-test results, we define selection criteria (Y-shift and X-shift) to determine RNA dependent proteins. Y-shift is the difference of protein amount between the global maxima of Ctrl and RNase group. X-shift is the difference of locations of global maxima (fraction) between the global maxima of Ctrl and RNase group. 

If the X- and Y-shift has positive values (left and down shift), we define the protein as RNA dependent. If the X- and Y-shift has negative values or close to zero (right and up shift), we define the protein as RNA independent. These shifts will be our second criteria for RNA dependency. We also combined our X- and Y-shifts in a dataframe and used the elbow method to determine the optimal number of clusters.	



2.6.3 T-Test of local maxima:

To find more RNA dependent proteins, we will include a 3rd criteria to classify the protein as RNA dependent. The 3rd criteria is the result of t-test of local maxima. 


2.7 Comparison with other databanks:

In order to calculate the true positive and true negative rate and see how good our criteria find the RNA dependent proteins we compare our finding with mammalian RNA-binding protein resources (https://r-deep.dkfz.de)



2.8 Linear regression:
Finally, we perform a linear regression. This model can generally be used to model the relationship between a dependent variable (regressand) and one or more explanatory variables (predictors). The relationship is represented by a linear function. We can see from the model to what extent the dependent variable can be explained by the other variables. 
Our regression model predicts the Y-Shift values with the information from correlation between Ctrl and RNase.

2.9 Working with complementary data:









3. Results 

3.1 Dataset exploration:


For further analysis of the replicates we sum up the total protein amount of all genes per replicate and plot them side by side in a bar plot. 
The bar plots revealed that there is a difference between the replicates and also there is a huge difference between Ctrl and RNase samples. The three replicates of Ctrl and RNase should theoretically have the same total amount since these are the results of the same experiment, but due to human factor in the lab working, the irregularities are not avoidable. 
In order to make the Ctrl and RNase comparable a normalization is needed.  



3.2 Fractionwise Normalization:

After we run the fractionwise normalization the protein amount between the three replicates of RNase dataframe and the three replicates of the Ctrl dataframe is equal. This can be seen in the barplot. The barplot also shows that although the protein amount of all three replicates of Ctrl and RNase is equal, there is a difference between the total amount of Ctrl ans RNase, which is exactly what we need. 

```{r}
##barplot einfügen
```


3.3 Anti outlier function:

The comparison of the MICA_HUMAN_Ctrl before and after the application of the anti_outlier_function has shown that the function reduced the outliers. Now the three replicates are comparable. 

```{r}
# plots einfügen
```


3.4 Peaks Identification: 

After the apply our self-written maximafunction the global (absolute) and local maxima are detected and the values for different thresholds are saved in different dataframes, for each threshold for RNase and Ctrl. 

In order to decide which threshold is the best for the local maxima we run our maxnum_plot_col function, which plots random protein with threshold in x-axis and number of maxima in y-axis. Having run the function several times,we decided to use a threshold value of 40% of the global maxima. Our chosen threshold of 40% protein value is enough to get only significant maxima and reduce the effect of noisy data. 


```{r}
# von Greta gemaltes Bild einfügen
```



3.5 Criteria for RNA dependency:


3.5.1 T-Test of global and local maxima:

A data frame **test** with 3680 rows and 1 column is produced after performing t-test of global maxima between Ctrl and RNase group. The row represents the name of the proteins that is alphabetically ordered. The only column represents the RNA dependency with 2 kind of results, TRUE or FALSE.

For t test of local maxima, the result is a data frame **test_0.4** that consists of 3680 rows and 1 column. The name of the proteins are in rows and the only column shows the RNA dependency. 3 outcomes are generated, TRUE, FALSE or NA.

```{r}
#Tabelle einfügen mit den TRUE/FALSE Ergebnissen 
```


3.5.2 K-Means Clustering (Y-shift and X-shift detection):

The elbow method revealed that the optimal number of clusters is three: RNA-dependent proteins cluster (X-shift and Y-shift are positive), RNA-independet proteins cluster (X-shift and Y-shift are negative) and not defined (no shift), or only one positive shift (X-shift or Y-shift). 

```{r}
#Bilder Cluster und elbow Methode 
```


3.5.3  Comparison of RNA dependency of both T-Test and K-Means:

We created a match function in order to find the proteins that fulfill the criteria for RNA-dependency, the T-test results for significant difference of the global maxima for Ctrl and RNAse and the K means clustering (Y-shift and X-shift detection).
We found 305 proteins that fulfill one of the two criteria and 63 proteins that fulfill both criteria. 


3.7 Comparison with other databanks:

The comparison with the table of mammalian RNA-binding protein resources has shown that our applied criteria finds the RNA-dependent proteins with very low false positive rate, but very high false negative rate. 
Out of 63 identified RNA-dependent proteins that fulfill both criteria, 57 had a match with the comparable databank and for one criteria fulfilled out of 305 identified proteins 228 had a match. 
Unfortunately our applied criteria could not identify 1860 proteins which makes the false negative rate really high.

```{r}
#Tabelle mit true negative, true positive, false negative und false positive

```








Local maxima (third criteria)











Linear Regression model 

The first linear regression is performed between correlation in x axis and y shift in y axis. The linear equation from the regression is $y = 1690907x - 1184175$. The p value for the predictor variable y is 7.98e^-06^ and for x is 5.96e^-09^. The p value of our model is 5.962e^-09^. A more detailed overview of of the regression **summary(lm)** can be seen in the picture below.

```{r}
picture
```

The histogram of the residues shows that the residuals are between -2e^-07^ and 2e^-07^. The QQ plot of the theoretical and sample residues aligns with the QQ line except when the theoretical quantiles are below -2.5 and above 2.5. 

```{r}
picture
```

The last plot is a plot with actual value in x axis, predicted value in y axis and a linear line with intercept 0 and slope 1. The data points gather mostly on the place where the predicted and actual values are above 0. Below 0 there are only few points are near the linear line. Nonetheless, there are also some points that are very far from the linear line.


```{r}
picture
```

The second linear regression is performed between RBP2GOScore in x axis and x shift in y axis. The linear equation from the regression is $y = 0.069630x -0.368093$. The p value for the predictor variable y is 0.00285 and for x is 2e^-16^. The p value of our model is 2.2e^-16^. A more detailed overview of of the regression **summary(lm_4)** can be seen in the picture below.

```{r}
picture
```

The histogram of the residues shows that the residuals are between -10 and 20. The QQ plot of the theoretical and sample residues aligns with the QQ line except when the theoretical quantiles are below -1.5 and above 1. 

```{r}
picture
```

The last plot is a plot with actual value in x axis, predicted value in y axis and a linear line with intercept 0 and slope 1. The data points gather mostly near the linear line when the actual value is between -2 and 2 and the predicted value is around 0.5.


```{r}
picture
```



Discussion 

Normalization 





Peaks Identification






Conclusion 

True positive and true negative rate 


References


